pkg/cmd/variable/create/create_test.go

```diff
@@ -0,0 +1,346 @@
+package create
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/cli/cli/v2/internal/config"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmd/variable/shared"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/httpmock"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/google/shlex"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewCmdCreate(t *testing.T) {
+	tests := []struct {
+		name     string
+		cli      string
+		wants    shared.PostPatchOptions
+		stdinTTY bool
+		wantsErr bool
+	}{
+		{
+			name:     "invalid visibility",
+			cli:      "cool_variable --org coolOrg -v'mistyVeil'",
+			wantsErr: true,
+		},
+		{
+			name:     "invalid visibility",
+			cli:      "cool_variable --org coolOrg -v'selected'",
+			wantsErr: true,
+		},
+		{
+			name:     "repos with wrong vis",
+			cli:      "cool_variable --org coolOrg -v'private' -rcoolRepo",
+			wantsErr: true,
+		},
+		{
+			name:     "no name",
+			cli:      "",
+			wantsErr: true,
+		},
+		{
+			name:     "multiple names",
+			cli:      "cool_variable good_variable",
+			wantsErr: true,
+		},
+		{
+			name:     "visibility without org",
+			cli:      "cool_variable -vall",
+			wantsErr: true,
+		},
```

Comment:
Can we add a test when is a given repo is not part of an org?

Reply:
present [here](https://github.com/cli/cli/pull/6928/files#diff-46c866368ca6af2cc87931248dc7f31940497cc72d187f8271e8e362e3fbb564R124)